kanidmd/src/lib/audit.rs

```diff
@@ -10,31 +10,172 @@ use chrono::DateTime;
 use uuid::adapter::HyphenatedRef;
 use uuid::Uuid;
 
-#[macro_export]
+#[derive(Debug, Serialize, Deserialize)]
+pub enum LogTag {
+    AdminError,
+    AdminWarning,
+    AdminInfo,
+    RequestError,
+    Security,
+    SecurityAccess,
+    Filter,
+    FilterError,
+    FilterWarning,
+    Trace,
+}
+
+impl fmt::Display for LogTag {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            LogTag::AdminError => write!(f, "admin::error 🚨"),
```

Comment:
I dunno if this icons will help or will make some difficulties showing logs and parsing them.

Reply:
Parsing there are still unique string tags associated, and I found the icons helpful? None of this is set in stone though, but for me when looking at the full log it helped things to standout. I'm attaching two screenshots as an example. Notice how easy it is to spot the access control lines? 

![Screen Shot 2020-05-22 at 16 28 53](https://user-images.githubusercontent.com/271005/82638090-718f8c00-9c49-11ea-9926-2ec280abf44a.png)

![Screen Shot 2020-05-22 at 16 29 13](https://user-images.githubusercontent.com/271005/82638074-6c324180-9c49-11ea-9cf7-cc74b9f4e68c.png)

Reply:
I find them pretty cool, helps with visualization and with that format I think that they are easy to parse. So I love them.